test/install_test.go

```diff
@@ -743,85 +770,96 @@ func TestLogs(t *testing.T) {
 				knownErrorsRegex = knownProxyErrorsRegex
 			}
 
-			t.Run(name, func(t *testing.T) {
-				outputStream, err := TestHelper.LinkerdRunStream(
-					"logs", "--no-color",
-					"--control-plane-component", deploy,
-					"--container", container,
-				)
-				if err != nil {
-					testutil.AnnotatedErrorf(t, "error running command",
-						"error running command:\n%s", err)
-				}
-				defer outputStream.Stop()
-				// Ignore the error returned, since ReadUntil will return an error if it
-				// does not return 10,000 after 2 seconds. We don't need 10,000 log lines.
-				outputLines, _ := outputStream.ReadUntil(10000, 2*time.Second)
+			outputStream, err := TestHelper.LinkerdRunStream(
+				"logs", "--no-color",
+				"--control-plane-component", deploy,
+				"--container", container,
+			)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("error running command:\n%s", err))
+				continue
+			}
+			defer outputStream.Stop()
+			// Ignore the error returned, since ReadUntil will return an error if it
+			// does not return 10,000 after 2 seconds. We don't need 10,000 log lines.
+			outputLines, _ := outputStream.ReadUntil(10000, 2*time.Second)
+			if len(outputLines) == 0 {
+				// Retry one time for 30 more seconds, in case the cluster is slow to
+				// produce log lines.
+				outputLines, _ = outputStream.ReadUntil(10000, 30*time.Second)
 				if len(outputLines) == 0 {
-					// Retry one time for 30 more seconds, in case the cluster is slow to
-					// produce log lines.
-					outputLines, _ = outputStream.ReadUntil(10000, 30*time.Second)
-					if len(outputLines) == 0 {
-						testutil.AnnotatedErrorf(t, "no logs found for %s", name)
-					}
+					errs = append(errs, fmt.Errorf("no logs found for %s", name))
 				}
+			}
 
-				for _, line := range outputLines {
-					if errRegex.MatchString(line) {
-						if knownErrorsRegex.MatchString(line) {
-							// report all known logging errors in the output
-							t.Logf("Found known error in %s log: %s", name, line)
+			for _, line := range outputLines {
+				if errRegex.MatchString(line) {
+					if knownErrorsRegex.MatchString(line) {
+						// report all known logging errors in the output
+						okMessages = append(okMessages, fmt.Sprintf("found known error in %s log: %s", name, line))
+					} else {
+						if proxy {
+							okMessages = append(okMessages, fmt.Sprintf("found unexpected proxy error in %s log: %s", name, line))
```

Comment:
I see this is similar to the previous behavior, but any idea why we don't append to `errs` on unknown errors in the proxy?

Reply:
According to #2647 the proxy error list kept on growing and so was causing regular CI failures.